### PR TITLE
[OFFAPPS-1042] Use I18n'd strings within the LTA 

### PR DESCRIPTION
**Pull request by vaevictis**
## Description
This PR implements the code-level changes allowed by [That sister PR](https://github.com/zendesk/linked_ticket_app/pull/83).

## Tasks
- [x] Use new strings in app's code
- [x] DRY that puppy and extract a couple of smaller functions
- [x] Find edge case blowing up in my face
- [x] Write plenty more tests
- [x] Sneak in a fix for the email autocomplete UI part that I broke some time ago

## Implementation notes
While testing new strings, I found a new edge case out when trying to hide the linked data field (requirement *or* setting).
If one field is present in the currently visible form, but not in *any other* form, I would still try to hide it. 

As I have no way to know exactly which field is *currently* visible, I added an error catching to protect against `client.invoke(<the_field_id>.hide)` called on any invisible field.

## References
[JIRA ticket](https://zendesk.atlassian.net/browse/OFFAPPS-XXX)
[Invision design](https://google.com.au)

## Screenshots (if needed)
See the [translations PR](https://github.com/zendesk/linked_ticket_app/pull/83) to see the screenshots

## CCs
Anyone specific?
@zendesk/apps-migration

## Risks
* [low] The edge cases are starting to get really convoluted, but I've added a solid test belt so I believi we are pretty safe against regression.

### CODE REVIEW

File: spec/src/app_spec.js [hunk 1]
```diff
@@ -784,46 +785,112 @@ describe('App', () => {
...
+
+          expect(client.invoke).toHaveBeenCalledWith('ticketFields:custom_field_123.hide')
+        })
+      })
+
+      describe('when the form is not visible', () => {
+        it('handles failure gracefully', (done) => {
+          spyOn(client, 'invoke').and.returnValues(rejectedPromise, resolvedPromise)
+          appInstance.hideOrNotifyForField(formName, fieldIds, id).then(() => {
+            expect(client.invoke).toHaveBeenCalled()
```
Comment (yu-tian113) [inline spec/src/app_spec.js]: redundant? `toHaveBeenCalledWith` already covers this one.
Do you want to test the called times with `client.invoke.calls.count()`?
Or both `toHaveBeenCalledWith()` and `toHaveBeenCalledWith(some params)`

Comment (vaevictis) [reply spec/src/app_spec.js]: Oh yes! I’ve created new tests when I extracted smaller functions. Forgot to clean after myself

File: spec/src/app_spec.js [hunk 2]
```diff
@@ -784,46 +785,112 @@ describe('App', () => {
...
       })
 
-      it('sends no alert if requirement field is present in all forms', (done) => {
+      it('sends no alert if requirement field is present in each form', (done) => {
         spyOn(client, 'get').and.returnValue(Promise.resolve({'requirement:linked_data': {'requirement_id': 987654321}}))
 
         appInstance.hideAncestryField().then(() => {
           expect(client.invoke).not.toHaveBeenCalledWith(
             'notify',
+            `A ticket form is missing the linked data field.\nForm: "first form"\nMissing field ID: "987654321"`,
```
Comment (keitheous) [inline spec/src/app_spec.js]: Just a suggesting, what if we rearrange the description here and on line 769 to be

`The linked data field is missing within / in a ticket form`

**Comment by keitheous** [comment]
Gentle reminder about our conversation yesterday, 

Regarding app_spec.js lines
- 22 & 25, and
- 33 & 36
merged as constants? 😃 

The file changes look good to me so I'll go ahead and give you a green tick

**Comment by vaevictis** [comment]
@keitheous yep I had seen one of these comments and fixed it, but I missed the other one! Just went a bit further on both of them today :)
